unattended.sh

```diff
@@ -1,186 +1,804 @@
 #!/bin/sh
 
-function error_and_exit {
-  echo -e "$1"
-  exit 1
-}
+### Define all supported constants
+OPTIONS=":t:v:r:l:"
+declare -A SUPPORTED_LOG_LEVEL=([DEBUG]=0 [INFO]=1 [WARN]=2 [ERROR]=3)
+declare -A SUPPORTED_TOPOLOGY=([central]=1 [poller]=1)
+declare -A SUPPORTED_VERSION=([21.04]=1)
+declare -A SUPPORTED_REPOSITORY=([testing]=1 [unstable]=1 [stable]=1)
+default_timeout_in_sec=5
+script_short_name="$(basename $0)"
+default_ip=$(hostname -I | awk '{print $1}')
+###
+
+#Define default values
+
+passwords_file=/etc/centreon/generated.tobesecured         #File where the generated passwords will be temporaly saved
+tmp_passwords_file=$(mktemp /tmp/generated.XXXXXXXXXXXXXX) #Random tmp file as the /etc/centreon does not exist yet
+
+topology=${ENV_CENTREON_TOPOLOGY:-"central"}    #Default topology to be installed
+version=${ENV_CENTREON_VERSION:-"21.04"}        #Default version to be installed
+repo=${ENV_CENTREON_REPO:-"stable"}             #Default repository to used
+operation=${ENV_CENTREON_OPERATION:-"install"}  #Default operation to be executed
+runtime_log_level=${ENV_LOG_LEVEL:-"INFO"}      #Default log level to be used
+selinux_mode=${ENV_SELINUX_MODE:-"permissive"}  #Default SELinux mode to be used
+wizard_autoplay=${ENV_WIZARD_AUTOPLAY:-"false"} #Default the install wizard is not run auto
+central_ip=${ENV_CENTRAL_IP:-$default_ip}       #Default central ip is the first of hostname -I
+
```

Comment:
a usage could be useful for a -h or by default when the user trigger the wrong option